runtime/domain.c

```diff
@@ -1713,7 +1713,17 @@ static void domain_terminate (void)
       caml_plat_broadcast(&s->cond);
       caml_plat_unlock(&s->lock);
 
+      /* signal the domain termination to the backup thread
+         NB: for a program with no additional domains, the backup thread
+         will not have been started */
+      atomic_store_rel(&domain_self->backup_thread_msg, BT_TERMINATE);
+      caml_plat_signal(&domain_self->domain_cond);
+      caml_plat_unlock(&domain_self->domain_lock);
```

Comment:
This is releasing the `domain_self->domain_lock` too early compared to `trunk`. Should we leave the `unlock` where it was and only move the other two lines here?

Reply:
Also, I would like to recreate this failure under `rr` and make sure that the failure is indeed due to the reason we think it is.